# Day-27 Assignment :

In [1]:
import pandas as pd

from sklearn import preprocessing
lab_enco = preprocessing.LabelEncoder()

from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score
from sklearn.metrics import confusion_matrix

from sklearn import neighbors
from sklearn import svm

In [2]:
data = pd.read_csv("train.csv")
data = data.drop(['PassengerId','Name','Cabin','Ticket'],axis=1)
data['Sex'] = lab_enco.fit_transform(data['Sex'])
data['Embarked'] = lab_enco.fit_transform(data['Embarked'])
data.head(2)

Survived  Pclass  Sex   Age  SibSp  Parch     Fare  Embarked
0         0       3    1  22.0      1      0   7.2500         2
1         1       1    0  38.0      1      0  71.2833         0

# Question-1 :-

#### Function:

In [3]:
def Knn(k,X,y):
    knn = neighbors.KNeighborsClassifier(n_neighbors=k)
    Score = knn.fit(X_train,y_train).score(X_test,y_test)
    return Score

In [4]:
y = data['Pclass']
X = data.drop(['Pclass'],axis=1)
X_train,X_test,y_train,y_test = train_test_split(X,y,test_size=0.3,random_state = 0)
k_value = 0
High_acc_Score = 0
for i in range(1,268):
    present_score = Knn(i,X,y)
    if(k_value == 0):
        k_value = i
        High_acc_Score = present_score
    if(present_score > High_acc_Score):
        k_value = i
        High_acc_Score = present_score

        
kn = neighbors.KNeighborsClassifier(n_neighbors=k_value)
y_pred = kn.fit(X_train,y_train).predict(X_test)
mat = confusion_matrix(y_test,y_pred)
print("\nHighest Accuracy = ",High_acc_Score*100,"%  and it is when  k =",k_value,"\n")
print("and Confusion matrix is :")
print(mat,"\n\n")




Highest Accuracy =  89.13857677902621 %  and it is when  k = 1 

and Confusion matrix is :
[[ 62   6   2]
 [  3  37   9]
 [  1   8 139]] 




# Question-2 :-

In [5]:
clff = svm.SVC(gamma=0.01,C=100)

In [6]:
def Svm(X,y):
    X_train,X_test,y_train,y_test = train_test_split(X,y,test_size=0.3,random_state = 0)
    y_pred = clff.fit(X_train,y_train).predict(X_test)
    acc_score = accuracy_score(y_test,y_pred,normalize=True)
    conf_mat = confusion_matrix(y_test,y_pred)
    return acc_score,conf_mat

In [7]:
depVars = ['Survived','Pclass','SibSp','Sex','Parch','Embarked']
for i in depVars:
    print("_____________________________________________________________________________________________________")
    print("  ",i,"  vs  Remmaining :-")
    print("_____________________________________________________________________________________________________\n")
    y = data[i]
    X = data.drop(i,axis=1)
    Score,Matrix =  Svm(X,y)
    print("Accuracy Score for", i ,"vs Remmaining  = ",Score*100,"%\n")
    print("And Confusion Matrix is :-")
    print(Matrix,"\n\n")

_____________________________________________________________________________________________________
   Survived   vs  Remmaining :-
_____________________________________________________________________________________________________

Accuracy Score for Survived vs Remmaining  =  74.15730337078652 %

And Confusion Matrix is :-
[[124  33]
 [ 36  74]] 


_____________________________________________________________________________________________________
   Pclass   vs  Remmaining :-
_____________________________________________________________________________________________________

Accuracy Score for Pclass vs Remmaining  =  90.2621722846442 %

And Confusion Matrix is :-
[[ 64   4   2]
 [  4  36   9]
 [  1   6 141]] 


_____________________________________________________________________________________________________
   SibSp   vs  Remmaining :-
_____________________________________________________________________________________________________

Accuracy Score for SibSp vs Remmai

### Pclass vs Remmaining combination has the highest Accuracy percentage i.e 90.26%

## --------------------------------------------------------------------------------------------------------------------